plugins/node/opentelemetry-instrumentation-pg/src/utils.ts

```diff
@@ -289,3 +290,68 @@ export function patchClientConnectCallback(
     cb.call(this, err);
   };
 }
+
+// Simplified W3CTraceContext serialization
+const W3C_TRACE_CONTEXT_VERSION = '00';
+const TRACE_PARENT_HEADER = 'traceparent';
+const TRACE_STATE_HEADER = 'tracestate';
+
+function hasValidSqlComment(query: string): boolean {
+  const indexOpeningDashDashComment = query.indexOf('--');
+  if (indexOpeningDashDashComment >= 0) {
+    return true;
+  }
+
+  const indexOpeningSlashComment = query.indexOf('/*');
+  if (indexOpeningSlashComment < 0) {
+    return false;
+  }
+
+  const indexClosingSlashComment = query.indexOf('*/');
+  return indexOpeningDashDashComment < indexClosingSlashComment;
+}
+
+export function addSqlCommenterComment(span: Span, query: string): string {
+  const spanContext = span.spanContext();
+
+  if (!isSpanContextValid(spanContext)) {
+    return query;
+  }
+
+  if (query.length === 0) {
+    return query;
+  }
+
+  // As per sqlcommenter spec we shall not add a comment if there already is a comment
+  // in the query
+  if (hasValidSqlComment(query)) {
+    return query;
+  }
+
+  const headers: { [key: string]: string } = {
```

Comment:
Is there a reason why you built the headers by yourself instead of using `W3CTraceContextPropagator`?

Reply:
I was unsure whether I should add a dependency against opentelemetry/core (where that propagator comes from), felt like it would be a bit too big of a dep? If that was not an issue, could definitely do that.

Reply:
core is quite small so I think it should be ok. It's likely installed anyway because the SDK depends on it.
The implementation there considers a bit more like the suppress propagation entry in context. This is likely a good thing but should not be repeated here and all other sql instrumentations.

Reply:
Refactored to use the propagator, as you mention that core is relatively small, I added it as a direct dependency instead of peer-dep.